doc/manual/source/building.rst

```diff
@@ -112,6 +112,33 @@ Then update the ``config.toml`` to use the appropriate paths.
 Updating
 """"""""
 
+.. danger::
+
+   In its current alpha version form Cascade will likely report errors if a
+   newer version is started using existing state and policy files created by
+   an older version.
+
+   Before upgrading delete all state and policy files. To enable you to keep
+   using existing signing keys after upgrade, the instructions below backup
+   your on-disk key files. Keys stored in HSMs are not affected by these
+   instructions. After upgrade you will need to use the ``--import-xxx``
+   options to the ``cascade zone add`` command to use them with any zones that
+   you (re)create.  will delete signing keys stored on disk**. If HSM signing
```

Comment:
This is grammatically wrong, but I don't know what exactly you want to write here.

Reply:
I thought I already corrected that, but after some discussion I'm going to revert the instructions about preserving keys anyway.